src/main/java/com/conveyal/datatools/manager/jobs/ProcessSingleFeedJob.java

```diff
@@ -42,20 +66,110 @@ public String getFeedVersionId () {
 
     @JsonProperty
     public String getFeedSourceId () {
-        return feedVersion.parentFeedSource().id;
+        return feedSource.id;
     }
 
+    /**
+     * The primary logic in this job handles loading (into Postgres) and validating the incoming GTFS file. However,
+     * there are important secondary functions that run {@link ArbitraryTransformJob} to modify either the input GTFS
+     * zip file or the resulting database representation.
+     *
+     * There are a few different transformation possibilities:
+     * 1. We only have zip transformations. These can either apply directly to the imported zip
+     *    or they can apply to a cloned version of the import. In the second case, the retrieval method should be set to
+     *    VERSION_CLONE (otherwise, it can be set to any other combination of retrieval methods).
+     * 2. We only have db transformations. These will always apply to a cloned snapshot of a feed version to
+     *    prevent the feed version's namespace from having different contents from the zip file. Following this, the
+     *    snapshot can either end up hanging (i.e., not published) or once all of the transformations have been
+     *    applied, we can create a new feed version from the snapshot (retrieval method being produced in house??).
+     * 3. Let's say we have zip transformations and db transformations.
+     *    a. Zip transformations apply to imported version, db transformations apply to snapshot or get published (no big deal)
+     *    b. Zip transformations apply to cloned version, db transformations apply to snapshot or get published.
+     *       This case is a little trickier, because we could end up with two cloned_versions. Let's say we want
+     *       a SQL query to run on the original data, create a new GTFS file, and then replace a GTFS+ file from some
+     *       other version. FTR1 = clone
```

Comment:
Still don't know what `FTR1 = clone` means.

It would probably be much better to visualize this with a flow chart. Either an ASCII comment indicating at what stage certain transformations might occur or a link to docs that has a diagram of boxes and lines indicating all this.

Reply:
I don't know it means either. Sorry, I meant to remove that mysterious comment.